extensions/src/main/java/org/pentaho/platform/web/http/api/resources/services/UserRoleDaoService.java

```diff
@@ -233,14 +242,21 @@ private ValidationFailedException validatePasswordFormat( String password ) {
       validationCriteria.add( Messages.getInstance().getString( "UserRoleDaoService.PassValidationError_SpecChar" ) );
     }
 
+    if ( !allowedCharsMatcher.matches() ) {
+      validationCriteria.add( Messages.getInstance().getString( "UserRoleDaoService.PassValidationError_PermittedChars" ) );
+    }
+
     errorMsg = errorMsg + String.join( ", ", validationCriteria ) + ".";
 
-    if ( ( password.length() < reqPassLength ) || ( isSpecCharReq && !password.matches( PASSWORD_SPEC_CHAR_PATTERN ) ) ) {
-      exception = new ValidationFailedException( errorMsg );
-      return exception;
-    } else {
+    if ( ( password.length() >= reqPassLength )
+      && allowedCharsMatcher.matches()
+      && ( isSpecCharReq && specCharsMatcher.matches() ) ) {
```

Comment:
This should be:
```suggestion
    if ( ( password.length() >= reqPassLength )
      && allowedCharsMatcher.matches()
      && ( !isSpecCharReq || specCharsMatcher.matches() ) ) {
```
Unit tests are likely failing due to this